addons/iot_drivers/tools/system.py

```diff
@@ -19,10 +36,304 @@
+    command = [*python_executable, 'pip', *args]
+
+    if IS_RPI and args[0] == 'install':
+        command.append('--user')
+        command.append('--break-system-package')
+
+    p = subprocess.run(command, stdout=subprocess.PIPE, check=False)
+    return p.returncode
+
+
+@cache
+def get_version(detailed_version=False):
+    if IS_RPI:
+        image_version = read_file_first_line('/var/odoo/iotbox_version')
+    elif IS_WINDOWS:
+        # updated manually when big changes are made to the windows virtual IoT
+        image_version = '23.11'
+    else:
+        image_version = 'test'
+
+    version = IOT_CHAR + image_version
+    if detailed_version:
+        # Note: on windows IoT, the `release.version` finish with the build date
+        version += f"-{odoo_release.version}"
+        if IS_RPI:
+            commit_hash = git("rev-parse", "--short", "HEAD")
+            version += f'#{commit_hash or "unknown"}'
+
+    return version
+
+
+def get_img_name():
+    major, minor = get_version()[1:].split('.')
+    return 'iotboxv%s_%s.zip' % (major, minor)
+
+
+def check_image():
+    """Check if the current image of IoT Box is up to date
+
+    :return: dict containing major and minor versions of the latest image available
+    :rtype: dict
+    """
+    try:
+        response = requests.get('https://nightly.odoo.com/master/iotbox/SHA1SUMS.txt', timeout=5)
+        response.raise_for_status()
+        data = response.content.decode()
+    except requests.exceptions.HTTPError:
+        _logger.exception('Could not reach the server to get the latest image version')
+        return False
+
+    current, latest = '', ''
+    hashes = {}
+    for line in data.splitlines():
+        if not line.strip():
+            continue
+        value, name = line.split('  ')
+        hashes[value] = name
+        if name == 'iotbox-latest.zip':
+            latest = value
+        elif name == get_img_name():
+            current = value
+    if current == latest:
+        return False
+
+    version = (
+        hashes.get(latest, 'Error')
+        .removeprefix('iotboxv')
+        .removesuffix('.zip')
+        .split('_')
+    )
+    return {'major': version[0], 'minor': version[1]}
+
+
+def update_conf(values, section='iot.box'):
+    """Update odoo.conf with the given key and value.
+
+    :param dict values: key-value pairs to update the config with.
+    :param str section: The section to update the key-value pairs in (Default: iot.box).
+    """
+    _logger.debug("Updating odoo.conf with values: %s", values)
+    conf = get_conf()
+
+    if not conf.has_section(section):
+        _logger.debug("Creating new section '%s' in odoo.conf", section)
+        conf.add_section(section)
+
+    for key, value in values.items():
+        conf.set(section, key, value) if value else conf.remove_option(section, key)
+
+        with open(path_file("odoo.conf"), "w", encoding='utf-8') as f:
+            conf.write(f)
```

Comment:
Why does this open, write, and close the config file for each entry being written?